webpack/dev.babel.js

```diff
@@ -24,32 +29,96 @@ const devConfig = configBuilder(
         "./src/standalone/index.js",
       ],
       "swagger-ui": "./src/style/main.scss",
+      vendors: ["react", "react-dom", "react-refresh/runtime"],
     },
 
     performance: {
       hints: false
     },
 
     output: {
-      library: "[name]",
       filename: "[name].js",
       chunkFilename: "[id].js",
+      library: {
+        name: "[name]",
+        export: "default",
+      },
+      publicPath: "/",
     },
 
     devServer: {
-      port: 3200,
-      publicPath: "/",
-      disableHostCheck: true, // for development within VMs
-      stats: {
-        colors: true,
+      allowedHosts: "all", // for development within VMs
+      headers: {
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": "*",
+        "Access-Control-Allow-Headers": "*",
       },
-      hot: true,
-      contentBase: path.join(__dirname, "../", "dev-helpers"),
+      port: 3200,
       host: "0.0.0.0",
+      hot: true,
+      static: {
+        directory: path.resolve(projectBasePath, "dev-helpers"),
+        publicPath: "/",
+      },
+      client: {
+        logging: "info",
+        progress: true,
+      },
+    },
+
+    module: {
+      rules: [
+        {
+          test: /\.jsx?$/,
+          include: [
+            path.join(projectBasePath, "src"),
+            path.join(projectBasePath, "node_modules", "object-assign-deep"),
+          ],
+          loader: "babel-loader",
+          options: {
+            retainLines: true,
+            cacheDirectory: true,
+            plugins: [isDevelopment && require.resolve("react-refresh/babel")].filter(Boolean),
+          },
+        },
+        {
+          test: /\.(txt|yaml)$/,
+          type: "asset/source",
+        },
+        {
+          test: /\.(png|jpg|jpeg|gif|svg)$/,
+          type: "asset/inline",
+        },
+        {
+          test: /\.(woff|woff2)$/,
+          type: "asset",
+        },
+        {
+          test: /\.(ttf|eot)$/,
+          type: "asset/resource",
+        },
+        {
+          test: /\.html$/,
+          use: [
+            {
+              loader: "html-loader",
+            }
+          ]
+        },
+      ],
     },
 
-    plugins: [new HotModuleReplacementPlugin()],
-  }
+    plugins: [
+      isDevelopment && new ReactRefreshWebpackPlugin({ library: "[name]" }),
+      new HtmlWebpackPlugin({
+        template: path.join(projectBasePath, "dev-helpers", "index.html"),
+      })
```

Comment:
We need to use `html-webpack-skip-assets-plugin` to skip including the `swagger-ui.js` file which is a byproduct of compiling the SASS files.

```suggestion
      }),
      new HtmlWebpackSkipAssetsPlugin({
        skipAssets: [/swagger-ui\.js/],
      }),
```

Reply:
ok, implemented.